### PR TITLE
update abstract tags for new labels

### PR DESCRIPTION
**Pull request by nonieA**
- cleaning newly labled data from EPMC_relevant_pubs_query2_manual_edit and _lizadditions .
- Refactoring of processing code
- Including codes from each 3 sources in final training data file too (might come in useful)
- Updating training code and experiment training notebook

We've gone from 859 tags to 1281 🎉 

| Tag code | Meaning | Number of grants |
|---|---|--- | 
| 1 | Relevant | 292 |
| 0 | Not relevant | 989 |

the output now looks a bit like (plus all the other columns from before): 
Relevance code | Normalised code - RF | Normalised code - grants | Normalised code - EPMC
-- | -- | -- | --
1 | 3 |   | 1
1 | 3 |   | 2
1 | 3 |   |  
1 | 3 |   |  
1 | 2 | 5 |  
1 | 2 | 5 |  
1 | 1 | 1 |  
1 | 1 |   |  
1 | 1 |   |  
1 | 1 |   |  
0 |   | 5 |  
0 |   | 5 |  
0 |   | 5 |  
0 |   | 5 |  

So it'll tell you what the original tags were and the overal 'relevance code' :
 If there is conflict in tags, the order of truth is:
    # 1. EPMC (this shows solid evidence of the output)
    # 2. RF (they might not have created the output at the time of the report)
    # 3. Grants (they might not have created the output at the time of the application)

# Results
1. Human accuracy on EPMC tags: 

Proportion of times we agree on relevant/not relevent:
0.8235294117647058
```
             Liz tag 3  Liz tag 1  Liz tag 2  Liz tag 5
Nonie tag 3          0          1          1          0
Nonie tag 1          0          9          1          0
Nonie tag 2          0          0          0          0
Nonie tag 5          0          0          0         30
```
2. Human accuracy on grant tags:

Proportion of times there was agreement
1.0
```
               Liz tag 1.0  Liz tag 5.0
Nonie tag 1.0            4            0
Nonie tag 5.0            0           23
```

3. Test data using TFIDF trained model:
```
              precision    recall  f1-score   support
           0       0.94      0.81      0.87        83
           1       0.80      0.94      0.87        70
    accuracy                           0.87       153
   macro avg       0.87      0.88      0.87       153
weighted avg       0.88      0.87      0.87       153
```
```
              predicted tag 0  predicted tag 1
actual tag 0               67               16
actual tag 1                4               66
```

4. Test data using Count trained model:
```
              precision    recall  f1-score   support
           0       0.90      0.96      0.93       219
           1       0.86      0.68      0.76        74
    accuracy                           0.89       293
   macro avg       0.88      0.82      0.84       293
weighted avg       0.89      0.89      0.89       293
```
```
              predicted tag 0  predicted tag 1
actual tag 0              211                8
actual tag 1               24               50
```

## Becky's labels
```
------Evaluation for Liz/Nonie and Becky's grants tags------
Nonie/Liz labelling of grant data set
1131
Merged code
1.0     111
4.0      16
5.0    1004
Becky labelling
76
Becky code
1.0    49
5.0    27
Proportion of times there was agreement
0.618421052631579
                   Becky tag 1.0  Becky tag 4.0  Becky tag 5.0
Nonie/Liz tag 1.0             28              0              2
Nonie/Liz tag 4.0             10              0              6
Nonie/Liz tag 5.0             11              0             19
```

We do worse on agreeing about the EPMC data, but it looks like a big cause is that Becky is more generous at labelling things as models than we were:
```
------Evaluation for Liz/Nonie and Becky's EPMC tags------
Nonie/Liz labelled:
Merged code
1     64
2     43
3     17
4     14
5    924
6     49
Becky labelled:
Becky code
1    11
2    24
3     8
5    16
6     1
Proportion of times we exactly agree on tool, dataset, model, not relevant:
0.4166666666666667
                 Becky tag 5  Becky tag 2  Becky tag 3  Becky tag 1
Nonie/Liz tag 5            9           15            2            4
Nonie/Liz tag 2            3            5            2            0
Nonie/Liz tag 3            4            2            4            0
Nonie/Liz tag 1            1            2            0            7
Proportion of times we agree on relevant/not relevent:
0.5166666666666667
                 Becky tag 5  Becky tag 2  Becky tag 1  Becky tag 3
Nonie/Liz tag 5            9            0            0            0
Nonie/Liz tag 2            0            5            0            2
Nonie/Liz tag 1            0            2            7            0
Nonie/Liz tag 3            0            2            0            4
```

### CODE REVIEW

File: nutrition_labels/grant_data_processing.py [hunk 1]
```diff
@@ -1,100 +1,190 @@
...
+    # Normalising the codes so they are all similar for different data sources
+    # 'None' if you dont want to include these tags in the training data
+    epmc_code_dict = {'1': 1, '2': 2, '3': 3, '4': None, '5': None, '6': None}
+    rf_code_dict = {'1': 1, '2': 2, '3': 3, '4': None, '5': None}
+    grants_code_dict = {'1': 1, '4': None, '5': 5}
+
+    # Process each of the 3 data sources separately and output a 
+    # dataframe for each of grant numbers - cleaned tags links
+    epmc_df = process_epmc(epmc_tags_query_one, epmc_tags_query_two, epmc_code_dict)
+    rf_df = process_RF(rf_tags, rf_code_dict)
```
Comment (nonieA) [inline nutrition_labels/grant_data_processing.py:127]: this seems to be breaking because of '55' in the codes did you clean them out before hand?

Comment (lizgzil) [reply nutrition_labels/grant_data_processing.py:127]: oops yes i did in the excel sheet. I'll upload the cleaned versions to sharepoint

Comment (lizgzil) [reply nutrition_labels/grant_data_processing.py:127]: i've added the cleaned RF data. let me know if it all works when using that (i think the other files should be ok)
